test/threadstest.h

```diff
@@ -65,6 +65,7 @@ static void *thread_run(void *arg)
     *(void **) (&f) = arg;
 
     f();
+    OPENSSL_thread_stop();
```

Comment:
Why is this needed? Don't we detect the thread stopping automatically anyway?

Reply:
I'm digging back in my mental history here, but IIRC there was a problem with the thread_run mechanism that required it.  I'll have to go back and look at the details though

Reply:
I took this example from the DRBG test, but yeah, it appears not needed

Reply:
Actually, I take that back, if I remove the thread_stop here I get memory leaks in the threads_test because the thread local data is never freed.